tests/samplers_tests/test_samplers.py

```diff
@@ -296,6 +295,62 @@ def objective(trial, exception):
     assert optuna.samplers.intersection_search_space(study) == {}
 
 
+def test_intersection_search_space_class() -> None:
+    search_space = optuna.samplers.IntersectionSearchSpace()
+    study = optuna.create_study()
+
+    # No trial.
+    assert search_space.calculate(study) == {}
+
+    # First trial.
+    study.optimize(lambda t: t.suggest_uniform("y", -3, 3) + t.suggest_int("x", 0, 10), n_trials=1)
+    assert search_space.calculate(study) == {
+        "x": IntUniformDistribution(low=0, high=10),
+        "y": UniformDistribution(low=-3, high=3),
+    }
+
+    # Returning sorted `OrderedDict` instead of `dict`.
+    assert search_space.calculate(study, ordered_dict=True) == OrderedDict(
+        [
+            ("x", IntUniformDistribution(low=0, high=10)),
+            ("y", UniformDistribution(low=-3, high=3)),
+        ]
+    )
+
+    # Second trial (only 'y' parameter is suggested in this trial).
+    study.optimize(lambda t: t.suggest_uniform("y", -3, 3), n_trials=1)
+    assert search_space.calculate(study) == {"y": UniformDistribution(low=-3, high=3)}
+
+    # Failed or pruned trials are not considered in the calculation of
+    # an intersection search space.
+    def objective(trial, exception):
+        # type: (optuna.trial.Trial, Exception) -> float
+
+        trial.suggest_uniform("z", 0, 1)
+        raise exception
+
+    study.optimize(lambda t: objective(t, RuntimeError()), n_trials=1, catch=(RuntimeError,))
+    study.optimize(lambda t: objective(t, optuna.exceptions.TrialPruned()), n_trials=1)
+    assert search_space.calculate(study) == {"y": UniformDistribution(low=-3, high=3)}
+
+    # If two parameters have the same name but different distributions,
+    # those are regarded as different trials.
```

Comment:
I'm not sure what does the `different trials` mean.

Reply:
I think that it might be a typo of "different parameters" (I copied this part from the existing `test_intersection_search_space` test function).

Reply:
Fixed: https://github.com/optuna/optuna/pull/1142/commits/07b3904178b81dad85bb940e92e86cd93e794321